runtime/module.go

```diff
@@ -281,7 +282,7 @@ func ProvideTransientStoreService(key depinject.ModuleKey, app *AppBuilder) stor
 	return transientStoreService{key: storeKey}
 }
 
-func ProvideAppVersionModifier(app *AppBuilder) baseapp.AppVersionModifier {
+func ProvideAppVersionModifier(app *AppBuilder) app.AppVersionModifier {
```

Comment:
Actually, we shouldn't have the app implementing it and just return nil here. This would clean-up the api and x/upgrade skips the logic if `AppVersionModifier` is nil.